### PR TITLE
[2311][Rebase & FF] STM Support

### PR DESCRIPTION
**Pull request by makubacki**
## Description

Adds SMM Transfer Monitor (STM) infrastructure changes.

### Commit Summary

- MdeModulePkg/TpmMeasurementLibNull: Add MM_CORE_STANDALONE
- BaseTools: Add the GenStm C tool

- [ ] Impacts functionality?
- [ ] Impacts security?
- [ ] Breaking change?
- [ ] Includes tests?
- [ ] Includes documentation?

## How This Was Tested

- Build with GenStm
- Generate an aux file using the build rule
- Link the TpmMeasureLibNull instance to a Standalone MM Core module

## Integration Instructions

- No change is needed unless this support is needed
- Review the GenStm arguments in help output for more details

### CODE REVIEW

**Review by kenlautner — APPROVED** [review]

File: BaseTools/Source/C/GenStm/Makefile [hunk 1]
```diff
@@ -0,0 +1,17 @@
+## @file
+# Windows makefile for the GenStm C tool build.
+#
+# Copyright (c) Microsoft Corporation
```
Comment (kuqin12) [inline BaseTools/Source/C/GenStm/Makefile]: Is this really Microsoft Corp copyright?

File: MdePkg/Include/Register/Intel/StmApi.h [hunk 2]
```diff
@@ -18,7 +18,6 @@
 
 #pragma pack (1)
 
-// MU_CHANGE: Added SMM_REV_ID definition, according to STM spec
```
Comment (kuqin12) [inline MdePkg/Include/Register/Intel/StmApi.h:21]: Is this removal needed?

File: BaseTools/Conf/build_rule.template [hunk 3]
```diff
@@ -120,8 +120,28 @@
 # 2.26 - Rename OemData to ProductData.
 # 2.27 - Add Rust build support.
 # 2.28 - Remove Rust Incremental Build Support Hack
+# 2.29 - Add STM aux file build rule
 
-#!VERSION=2.28
+#!VERSION=2.29
+
+# MU_CHANGE [BEGIN] - Add STM aux file build rule
```
Comment (kuqin12) [inline BaseTools/Conf/build_rule.template]: @Javagedes Do we still need this?

**Review by kuqin12 — APPROVED** [review]

File: BaseTools/Conf/build_rule.template (hunk 3 above)
Comment (Javagedes) [reply BaseTools/Conf/build_rule.template]: no

File: BaseTools/Source/C/GenStm/Makefile (hunk 1 above)
Comment (makubacki) [reply BaseTools/Source/C/GenStm/Makefile]: It previously had no copyright, so I assumed it originated in our changes. I'll trace the lineage.

File: MdePkg/Include/Register/Intel/StmApi.h (hunk 2 above)
Comment (makubacki) [reply MdePkg/Include/Register/Intel/StmApi.h:21]: Yes, it's not cherry-picked from directly from in the first commit of the series.

**Comment by makubacki** [comment]
Will be a pushing a change for the GenStm AARCH64 BaseTools build failure soon.

File: BaseTools/Conf/build_rule.template (hunk 3 above)
Comment (makubacki) [reply BaseTools/Conf/build_rule.template]: I'll drop it.

File: BaseTools/Source/C/GenStm/Makefile (hunk 1 above)
Comment (makubacki) [reply BaseTools/Source/C/GenStm/Makefile]: Updated to Intel copyright based on code in https://github.com/jyao1/STM/blob/master/Stm/StmPkg/EdkII/BaseTools/Source/C/GenStm/Makefile

File: BaseTools/Conf/build_rule.template (hunk 3 above)
Comment (makubacki) [reply BaseTools/Conf/build_rule.template]: Dropped from latest push
